### PR TITLE
Deploy with interactive defaults

### PR DESCRIPTION
**Pull request by gerhard**
I had to deploy recently and would have liked sensible defaults with prompts. What do you think?

### CODE REVIEW

**Comment by michaelklishin** [comment]
I don't see much need for interactive inputs as we have only 2 required parameters. This PR produces no deployment command output for me either interactively or via `./deploy {username}@next.rabbitmq.com next`.

**Comment by michaelklishin** [comment]
That said, if positional arguments continue working just like today, why not. But currently I get no output from any `ssh` commands, so something is wrong.

Also, this PR was probably meant to be submitted against `live`.

**Comment by gerhard** [comment]
This PR does not change the current behaviour when using positional arguments, so it's fully backwards compatible.

The interactive nature and sensible defaults kick in when running just `./deploy`

**Comment by gerhard** [comment]
If I run it in debug mode (`DEBUG=1`) I get the following:

```sh
+ set -e
+ REMOTE_HOST=gerhard@www.rabbitmq.com
+ VERSION=www
+ VERSIONS=(previous www next)
+ GIT_SHA=
+ '[' -n '' ']'
+ echo

+ read -rp 'Which user & host do you want to deploy to? (press enter for gerhard@www.rabbitmq.com): ' remote_host
Which user & host do you want to deploy to? (press enter for gerhard@www.rabbitmq.com):
+ REMOTE_HOST=gerhard@www.rabbitmq.com
+ '[' -n '' ']'
+ echo

+ read -rp 'Which website version is this? (press enter for www or choose one of: previous www next): ' version
Which website version is this? (press enter for www or choose one of: previous www next):
+ VERSION=www
+ TMP_DIR=/tmp/to_deploy_www
+ ssh gerhard@www.rabbitmq.com mkdir -p /tmp/to_deploy_www
+ rsync -qcrz --delete --perms --chmod=a+rwX --exclude=site/man code conf site wordpress-theme gerhard@www.rabbitmq.com:/tmp/to_deploy_www
+ ssh -t gerhard@www.rabbitmq.com sudo '/tmp/to_deploy_www/code/deploy-remote www '
[sudo] password for gerhard:
Feeds will be mirrored by the user rabbitmq
######.#.#####.###########.##.#..#######.#####.##.###.#..##.#..#.##.##.####.##.#.########.#######...#####.#..###.##.###########.#.#########.######.###.#.##########.###.....######################################.#############.########.######....................................................................................................................................######........................................................................................................................................................#
Site www.rabbitmq.com already enabled
[ ok ] Reloading web server config: apache2.
Connection to www.rabbitmq.com closed.
```

File: deploy [hunk 1]
```diff
@@ -1,20 +1,36 @@
-#!/bin/sh
+#!/bin/bash
```
Comment (dumbbell) [inline deploy]: This is not portable :) Please use `/usr/bin/env bash` if you really want to use Bash.

File: deploy [hunk 2]
```diff
@@ -1,20 +1,36 @@
...
+
+[ -z "$DEBUG" ] || set -x
 
 set -e
 
-remote_host=$1
-mode=$2
-version=$3
+REMOTE_HOST="$USER@www.rabbitmq.com"
```
Comment (dumbbell) [inline deploy]: There is no need to prepend `$USER@`. ssh(1) will do that itself and you could also set the user in your ssh_config(5).

**Comment by dumbbell** [comment]
@michaelklishin has a point: the script in `live` calls `set -x` before starting the actual deploy. The new script should keep this behavior, even if you set it earlier with `DEBUG=1`. Otherwise, it asks for a password but we don't have any context.

**Comment by gerhard** [comment]
👀

**Comment by gerhard** [comment]
@michaelklishin 👀

**Pull request merged**